service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/XpathExtractor.java

```diff
@@ -0,0 +1,354 @@
+    private static final String MEDICATION_REQUEST_SECOND_PARTICIPANT = MEDICATION_REQUEST_AUTHOR_REF;
+
+    private static final String CONDITION = COMPONENT + LINKSET;
+    private static final String CONDITION_AVAILABILITY_TIME = CONDITION + AVAILABILITY_SELECTOR;
+    private static final String CONDITION_AUTHOR_TIME = CONDITION_AVAILABILITY_TIME;
+    private static final String CONDITION_EFFECTIVE_TIME_LOW = CONDITION + EFFECTIVE_TIME_LOW_SELECTOR;
+    private static final String CONDITION_EFFECTIVE_TIME_CENTER = CONDITION + EFFECTIVE_TIME_CENTER_SELECTOR;
+    private static final String CONDITION_AUTHOR_REF = CONDITION + PARTICIPANT_SELECTOR;
+    private static final String CONDITION_SECOND_PARTICIPANT = CONDITION_AUTHOR_REF;
+
+    private static final String PROCEDURE_REQUEST = COMPONENT + PLAN_STATEMENT;
+    private static final String PROCEDURE_REQUEST_AVAILABILITY_TIME = PROCEDURE_REQUEST + AVAILABILITY_SELECTOR;
+    private static final String PROCEDURE_REQUEST_AUTHOR_TIME = PROCEDURE_REQUEST_AVAILABILITY_TIME;
+    private static final String PROCEDURE_REQUEST_EFFECTIVE_TIME = PROCEDURE_REQUEST_AVAILABILITY_TIME;
+    private static final String PROCEDURE_REQUEST_AUTHOR_REF = PROCEDURE_REQUEST + PARTICIPANT_SELECTOR;
+    private static final String PROCEDURE_REQUEST_SECOND_PARTICIPANT = PROCEDURE_REQUEST_AUTHOR_REF;
+
+    private static final String DOCUMENT_REFERENCE = COMPONENT + NARRATIVE_STATEMENT;
+    private static final String DOCUMENT_REFERENCE_AVAILABILITY_TIME = DOCUMENT_REFERENCE + AVAILABILITY_SELECTOR;
+    private static final String DOCUMENT_REFERENCE_AUTHOR_TIME = DOCUMENT_REFERENCE_AVAILABILITY_TIME;
+    private static final String DOCUMENT_REFERENCE_EFFECTIVE_TIME = DOCUMENT_REFERENCE_AVAILABILITY_TIME;
+    private static final String DOCUMENT_REFERENCE_AUTHOR_REF = DOCUMENT_REFERENCE + PARTICIPANT_SELECTOR;
+    private static final String DOCUMENT_REFERENCE_SECOND_PARTICIPANT = DOCUMENT_REFERENCE_AUTHOR_REF;
+
+    private static final String AVAILABILITY_TIME_VALUE_TEMPLATE = "<availabilityTime value=\"%s\"/>";
+    private static final String EFFECTIVE_TIME_CENTER_TEMPLATE = "<center value=\"%s\"/>";
+    private static final String DEFAULT_TIME_VALUE = "<center nullFlavor=\"UNK\"/>";
+    private static final String DEFAULT_AVAILABILITY_TIME_VALUE = "<availabilityTime nullFlavor=\"UNK\"/>";
+    private static final String PARTICIPANT_TEMPLATE = "<Participant2 typeCode=\"PPRRF\" contextControlCode=\"OP\">%n"
+        + "           <agentRef classCode=\"AGNT\">%n"
+        + "               <id root=\"%s\"/>%n"
+        + "           </agentRef>%n"
+        + "       </Participant2>";
+    private static final String END_OF_LINKSET_COMPONENT = "</LinkSet>\n</component>";
+
+    private static final XPathService X_PATH_SERVICE = new XPathService();
+
+    public static EncounterTemplateParameters
+        .EncounterTemplateParametersBuilder extractValuesForUncategorizedObservation(String component) {
```

Comment:
I think importing the builder class directly would make this signatures a lot more readable